plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestInternalFieldConflict.java

```diff
@@ -0,0 +1,72 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.kafka.TestingKafka;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.kafka.util.TestUtils.createDescription;
+import static io.trino.plugin.kafka.util.TestUtils.createOneFieldDescription;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@Test(singleThreaded = true)
+public class TestInternalFieldConflict
+        extends AbstractTestQueryFramework
+{
+    private SchemaTableName topicWithDefaultPrefixes;
+    private SchemaTableName topicWithCustomPrefixes;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        TestingKafka testingKafka = closeAfterClass(TestingKafka.create());
+        topicWithDefaultPrefixes = new SchemaTableName("default", "test_" + randomTableSuffix());
+        topicWithCustomPrefixes = new SchemaTableName("default", "test_" + randomTableSuffix());
+        return KafkaQueryRunner.builder(testingKafka)
+                .setExtraTopicDescription(ImmutableMap.of(
+                        topicWithDefaultPrefixes, createDescription(
+                                topicWithDefaultPrefixes,
+                                createOneFieldDescription("_key", createVarcharType(15)),
+                                ImmutableList.of(createOneFieldDescription("custkey", BIGINT), createOneFieldDescription("acctbal", DOUBLE))),
+                        topicWithCustomPrefixes, createDescription(
+                                topicWithCustomPrefixes,
+                                createOneFieldDescription("unpredictable_prefix_key", createVarcharType(15)),
+                                ImmutableList.of(createOneFieldDescription("custkey", BIGINT), createOneFieldDescription("acctbal", DOUBLE)))))
+                .setExtraKafkaProperties(ImmutableMap.<String, String>builder()
+                        .put("kafka.internal-column-prefix", "unpredictable_prefix_")
+                        .buildOrThrow())
+                .build();
+    }
+
+    @Test
+    public void testInternalFieldPrefix()
+    {
+        assertQuery("SELECT count(*) FROM " + topicWithDefaultPrefixes, "VALUES 0");
+        assertThatThrownBy(() -> assertQuery("SELECT count(*) FROM " + topicWithCustomPrefixes)).getCause()
+                .hasMessage("Internal kafka columns names conflict with columns names from the topic. "
```

Comment:
What is the top-most Exception being thrown? Is it a TrinoException?

Reply:
No, it was not.
https://github.com/trinodb/trino/pull/14224#discussion_r986478030

Reply:
As I've modified it to `TrinoException`, it could be changed to assertQueryFails.